src/windows_utils.rs

```diff
@@ -1,222 +1,177 @@
-use std::{
-    ffi::CString,
-    ptr::{null, null_mut},
+use windows::core::*;
+use windows::Win32::Foundation::*;
+use windows::Win32::System::LibraryLoader::GetModuleFileNameA;
+use windows::Win32::System::Registry::{
+    RegCloseKey, RegCreateKeyExA, RegDeleteTreeA, RegSetValueExA, HKEY, HKEY_CURRENT_USER,
+    KEY_READ, KEY_WRITE, REG_OPTION_NON_VOLATILE, REG_SZ,
 };
-
-use winapi::{
-    shared::{
-        minwindef::{DWORD, HKEY, MAX_PATH},
-        windef::DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2,
-    },
-    um::{
-        libloaderapi::GetModuleFileNameA,
-        winnt::{KEY_WRITE, REG_OPTION_NON_VOLATILE, REG_SZ},
-        winreg::{RegCloseKey, RegCreateKeyExA, RegDeleteTreeA, RegSetValueExA, HKEY_CURRENT_USER},
-        winuser::SetProcessDpiAwarenessContext,
-    },
+use windows::Win32::UI::HiDpi::{
+    SetProcessDpiAwarenessContext, DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2,
 };
 
 use crate::error_msg;
 
-fn get_binary_path() -> String {
-    let mut buffer = vec![0u8; MAX_PATH];
+const REGISTRY_PATH_DIRECTORY: PCSTR =
+    s!("Software\\Classes\\Directory\\Background\\shell\\Neovide");
+const REGISTRY_PATH_DIRECTORY_COMMAND: PCSTR =
+    s!("Software\\Classes\\Directory\\Background\\shell\\Neovide\\command");
+const REGISTRY_PATH_FOLDER: PCSTR = s!("Software\\Classes\\*\\shell\\Neovide");
+const REGISTRY_PATH_FOLDER_COMMAND: PCSTR = s!("Software\\Classes\\*\\shell\\Neovide\\command");
+
+fn get_neovide_path() -> String {
+    let mut buffer = vec![0u8; MAX_PATH as usize];
+    let len: u32;
     unsafe {
-        GetModuleFileNameA(
-            null_mut(),
-            buffer.as_mut_ptr() as *mut i8,
-            MAX_PATH as DWORD,
-        );
-        CString::from_vec_unchecked(buffer)
-            .into_string()
-            .unwrap_or_else(|_| "".to_string())
-            .trim_end_matches(char::from(0))
-            .to_string()
+        len = GetModuleFileNameA(HMODULE::default(), &mut buffer);
     }
+    buffer.truncate(len as usize);
+    String::from_utf8(buffer).unwrap()
 }
 
 fn unregister_rightclick() -> bool {
-    let str_registry_path_1 =
-        CString::new("Software\\Classes\\Directory\\Background\\shell\\Neovide").unwrap();
-    let str_registry_path_2 = CString::new("Software\\Classes\\*\\shell\\Neovide").unwrap();
     unsafe {
-        let s1 = RegDeleteTreeA(HKEY_CURRENT_USER, str_registry_path_1.as_ptr());
-        let s2 = RegDeleteTreeA(HKEY_CURRENT_USER, str_registry_path_2.as_ptr());
-        s1 == 0 && s2 == 0
+        RegDeleteTreeA(HKEY_CURRENT_USER, REGISTRY_PATH_DIRECTORY).0 == 0
+            && RegDeleteTreeA(HKEY_CURRENT_USER, REGISTRY_PATH_FOLDER).0 == 0
     }
 }
 
 fn register_rightclick_directory() -> bool {
```

Comment:
It's more idiomatic to return a `Ressult<(), ()> here instead of a boolean indicating failure.